src/index.ts

```diff
@@ -420,13 +287,29 @@ export default function solidPlugin(options: Partial<Options> = {}): Plugin {
 
         return extensionOptions.typescript;
       });
+      const plugins: NonNullable<NonNullable<babel.TransformOptions['parserOpts']>['plugins']> = ['jsx']
 
       if (shouldBeProcessedWithTypescript) {
-        opts.presets.push([ts, options.typescript || {}]);
+        plugins.push('typescript');
       }
 
+      const opts: babel.TransformOptions = {
+        root: projectRoot,
+        filename: id,
+        sourceFileName: id,
+        presets: [[solid, { ...solidOptions, ...(options.solid || {}) }]],
+        plugins: needHmr && !isSsr && !inNodeModules ? [[solidRefresh, { bundler: 'vite' }]] : [],
+        ast: false,
+        sourceMaps: true,
+        configFile: false,
+        babelrc: false,
+        parserOpts: {
```

Comment:
Since Babel doesn't actually extend the language, TS is already built into its core. Babel can process TypeScript files by enabling it through the parser options.